Build of 22.0.5-Piers fails when building against “meson built tinyxml-11.0.0”

The following error occurs when trying to find tinyxml2 (note the Kodi build is happy to use a meson built tinyxml2)
 
```

UNPACK      peripheral.joystick
BUILD      peripheral.joystick (target)
    TOOLCHAIN      cmake (auto-detect)
Executing (target): cmake -GNinja -DCMAKE_TOOLCHAIN_FILE=/var/media/DATA/home-rudi/LibreELEC.kernel/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/etc/cmake-x86_64-libreelec-linux-gnu.conf -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=MinSizeRel /var/media/DATA/home-rudi/LibreELEC.kernel/build.LibreELEC-Generic.x86_64-13.0-devel/build/peripheral.joystick-22.0.5-Piers
-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is GNU 15.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /var/media/DATA/home-rudi/LibreELEC.kernel/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /var/media/DATA/home-rudi/LibreELEC.kernel/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Checking to see if CXX compiler accepts flag -flto
-- Checking to see if CXX compiler accepts flag -flto - yes
CMake Error at CMakeLists.txt:11 (find_package):
  By not providing "FindTinyXML2.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "TinyXML2",
  but CMake did not find one.

  Could not find a package configuration file provided by "TinyXML2" with any
  of the following names:

    TinyXML2Config.cmake
    tinyxml2-config.cmake

  Add the installation prefix of "TinyXML2" to CMAKE_PREFIX_PATH or set
  "TinyXML2_DIR" to a directory containing one of the above files.  If
  "TinyXML2" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
FAILURE: s/build peripheral.joystick during configure_target (default)
*********** FAILED COMMAND ***********
CMAKE_EXPORT_COMPILE_COMMANDS=1 cmake ${CMAKE_GENERATOR_NINJA} ${TARGET_CMAKE_OPTS} ${PKG_CMAKE_OPTS_TARGET} ${PKG_CMAKE_SCRIPT%/*}
**************************************

``` 
Meson built files:
```

$ find -type f
./.libreelec-package
./usr/lib/pkgconfig/tinyxml2.pc
./usr/lib/libtinyxml2.a
./usr/include/tinyxml2.h

``` 
Cmake built files
```
$ find -type f
./.libreelec-package
./usr/lib/cmake/tinyxml2/tinyxml2-config-version.cmake
./usr/lib/cmake/tinyxml2/tinyxml2-static-targets.cmake
./usr/lib/cmake/tinyxml2/tinyxml2-static-targets-release.cmake
./usr/lib/cmake/tinyxml2/tinyxml2-config.cmake
./usr/lib/pkgconfig/tinyxml2.pc
./usr/lib/libtinyxml2.a
./usr/include/tinyxml2.h
```

--- a/cmake/FindTinyXML2.cmake
+++ b/cmake/FindTinyXML2.cmake
@@ -1,0 +1,42 @@
+# FindTinyXML2
+# -----------
+# Finds the TinyXML2 library
+#
+# This will define the following variables:
+#
+# TINYXML2_FOUND - system has TinyXML2
+#
+# and the following imported targets:
+#
+#   tinyxml2::tinyxml2 - The UDEV library
+
+find_package(TinyXML2 CONFIG)
+
+if(TinyXML2_FOUND)
+  set(TINYXML2_FOUND TinyXML2_FOUND)
+elseif(NOT TinyXML2_FOUND)
+  find_package(PkgConfig)
+  if(PKG_CONFIG_FOUND)
+    pkg_check_modules(PC_TINYXML2 tinyxml2 QUIET)
+  endif()
+
+  find_path(TINYXML2_INCLUDE_DIR tinyxml2.h
+                                 PATHS ${PC_TINYXML2_INCLUDEDIR})
+  find_library(TINYXML2_LIBRARY NAMES tinyxml2
+                                PATHS ${PC_TINYXML2_LIBDIR})
+  set(TINYXML2_VERSION ${PC_TINYXML2_VERSION})
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(TinyXML2
+                                    REQUIRED_VARS TINYXML2_LIBRARY TINYXML2_INCLUDE_DIR
+                                    VERSION_VAR TINYXML2_VERSION)
+
+  if(TINYXML2_FOUND)
+    add_library(tinyxml2::tinyxml2 UNKNOWN IMPORTED)
+    set_target_properties(tinyxml2::tinyxml2 PROPERTIES
+                          IMPORTED_LOCATION "${TINYXML2_LIBRARY}"
+                          INTERFACE_INCLUDE_DIRECTORIES "${TINYXML2_INCLUDE_DIR}")
+  endif()
+
+  mark_as_advanced(TINYXML2_INCLUDE_DIR TINYXML2_LIBRARY)
+endif()

